server/store/user_store.go

```diff
@@ -12,11 +13,15 @@ import (
 )
 
 type UserStore interface {
-	LoadUser(mattermostUserId string) (*User, error)
-	LoadMattermostUserId(remoteUserId string) (string, error)
+	LoadUser(mattermostUserID string) (*User, error)
+	LoadMattermostUserID(remoteUserID string) (string, error)
 	LoadUserIndex() (UserIndex, error)
-	StoreUser(user *User) error
-	DeleteUser(mattermostUserId string) error
+	StoreNewUser(user *User) error
+	LoadUserFromIndex(mattermostUserID string) (*UserShort, error)
+	DeleteUser(mattermostUserID string) error
+	AtomicModifyUserIndex(modify func(userIndex UserIndex) (UserIndex, error)) error
```

Comment:
Why use `Atomic...` prefix in the names if there are no non-atomic equivalents to disambiguate from?